hazelcast/src/main/java/com/hazelcast/internal/util/JVMUtil.java

```diff
@@ -112,6 +113,19 @@ public static long usedMemory(final Runtime runtime) {
         return used;
     }
 
+    /**
+     * Explicit cast to {@link Buffer} parent buffer type. It resolves issues the covariant return types in Java 9+ for
+     * {@link java.nio.ByteBuffer} and {@link java.nio.CharBuffer}. Explicit casting resolves the NoSuchMethodErrors (e.g
+     * java.lang.NoSuchMethodError: java.nio.ByteBuffer.limit(I)Ljava/nio/ByteBuffer) when the project is compiled with newer
+     * Java version and run on Java 8.
+     *
+     * @param buf buffer to cast to the abstract {@link Buffer} parent type
+     * @return the provided buffer
+     */
+    public static Buffer useBuffer(Buffer buf) {
```

Comment:
What do you say to `upcast`? _naming is hard_

Reply:
Whatever :) The `upcast` sounds so royal, I'll like it :-P